src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -26,27 +26,53 @@ class ConfigurationHelper implements Serializable {
-        return mixin(stepConfiguration, filter)
+        return mixin(stepConfiguration, filter, step, compatibleParameters)
     }
 
-    ConfigurationHelper mixinStageConfig(commonPipelineEnvironment, stageName, Set filter = null){
+    ConfigurationHelper mixinStageConfig(commonPipelineEnvironment, stageName, Set filter = null, Script step = null, Map compatibleParameters = [:]){
         Map stageConfiguration = ConfigurationLoader.stageConfiguration([commonPipelineEnvironment: commonPipelineEnvironment], stageName)
-        return mixin(stageConfiguration, filter)
+        return mixin(stageConfiguration, filter, step, compatibleParameters)
     }
 
-    ConfigurationHelper mixinStepConfig(commonPipelineEnvironment, Set filter = null){
+    ConfigurationHelper mixinStepConfig(commonPipelineEnvironment, Set filter = null, Script step = null, Map compatibleParameters = [:]){
         if(!name) throw new IllegalArgumentException('Step has no public name property!')
         Map stepConfiguration = ConfigurationLoader.stepConfiguration([commonPipelineEnvironment: commonPipelineEnvironment], name)
-        return mixin(stepConfiguration, filter)
+        return mixin(stepConfiguration, filter, step, compatibleParameters)
     }
 
-    ConfigurationHelper mixin(Map parameters, Set filter = null){
+    ConfigurationHelper mixin(Map parameters, Set filter = null, Script step = null, Map compatibleParameters = [:]){
+        if (parameters.size() > 0 && compatibleParameters.size() > 0) {
+            parameters = ConfigurationMerger.merge(handleCompatibility(step, compatibleParameters, parameters), null, parameters)
+        }
         config = ConfigurationMerger.merge(parameters, filter, config)
         return this
     }
 
+    private Map handleCompatibility(Script step, Map compatibleParameters, String paramStructure = '', Map configMap ) {
+        Map newConfig = [:]
+        compatibleParameters.each {entry ->
+            if (entry.getValue() instanceof Map) {
+                paramStructure = (paramStructure ? paramStructure + '.' : '') + entry.getKey()
+                newConfig[entry.getKey()] = handleCompatibility(step, entry.getValue(), paramStructure, configMap)
+            } else {
+                def configSubMap = configMap
+                for(String key in paramStructure.tokenize('.')){
+                    configSubMap = configSubMap?.get(key)
+                }
+                if (configSubMap == null || (configSubMap != null && configSubMap[entry.getKey()] == null)) {
+                    newConfig[entry.getKey()] = configMap[entry.getValue()]
+                    def paramName = (paramStructure ? paramStructure + '.' : '') + entry.getKey()
+                    if (step) {
+                        step.echo ("[INFO] The parameter '${entry.getValue()}' is COMPATIBLE to the parameter '${paramName}'")
```

Comment:
This leads to a lot of log output which might be annoying. It is shown even if I use the new config parameters.. Maybe we can just output this information in case we really handle the compatibility.

Reply:
good point, I changed it in a way that the log output is only written if the old param value is filled